### ISSUE TITLE
SMTP: Container ID used as a domain name in EHLO violates RFC5321 

### ISSUE THREAD

**Issue by scibi** [getsentry/sentry#16386]
## Important Details

How are you running Sentry?

* [X] On-Premise docker [Version 10.1.0.dev0]
* [ ] Saas (sentry.io)
* [ ] Other [briefly describe your environment]

## Description

I've set up new Sentry installation as described here https://docs.sentry.io/server/installation/ on fresh Debian Buster system. I configured existing Postfix server running on another machine as a mail server. I pressed "Send test email" button on `https://sentry-fqdn/manage/status/mail/` and I've got an error message:

```
{u'my.email@address: (450, '4.7.1 <c93fcab98fcb>: Helo command rejected: Host not found')}
```
(BTW formatting is broken here)

After some debugging I discovered that this is caused by sending

```
EHLO c93fcab98fcb
```

Where `c93fcab98fcb` is a container ID od `sentry_onpremise_web_1`.

According to RFC 53212:

> The domain name given in the EHLO command MUST be either a primary host name (a domain name that resolves to an address RR) or, if the host has no name, an address literal, as described in Section 4.1.3 and discussed further in the EHLO discussion of Section 4.1.4.

So this clearly violates this RFC so emails are rejected by a SMTP server.

## Steps to Reproduce

1. Install Sentry using Docker
2. Configure it to use some SMTP server
3. Try to send test email
4. Snoop on the communication between Sentry and SMTP server

### What you expected to happen

It should be possible to specify domain used in ELHO command in configuration if it can't be autodetected.

### Possible Solution

Probably it's necessary to fix this in Django too because there is no way to pass value to EmailBackend (https://github.com/django/django/blob/master/django/core/mail/backends/smtp.py#L53)

**Comment by BYK** [comment]
I think you should just be able to set this value properly: https://github.com/getsentry/onpremise/blob/master/sentry/config.example.yml#L11

I was able to use a well-known SMTP server with Sentry on-premise so I don't think anything needs patching.

**Comment by scibi** [comment]
Hi!

`mail.host` is an address of SMTP server. Sentry makes a TCP connection to this server to port 25 (`mail.port`) and sends `EHLO container_id` as a first command. It should send `EHLO sentry.server.fqdn.com` instead.

**Comment by BYK** [comment]
>mail.host is an address of SMTP server. Sentry makes a TCP connection to this server to port 25 (mail.port) and sends EHLO container_id as a first command. It should send EHLO sentry.server.fqdn.com instead.

Ughh, yeah sorry. Not sure what I was thinking when I replied with that. I was actually thinking about this line but that's also different than what you are saying: https://github.com/getsentry/onpremise/blob/dffd59723c05eb7a8a13f441a6cbb8b4e85fc437/sentry/config.example.yml#L17

This seems related to https://code.djangoproject.com/ticket/6989

We _can_ introduce a patch or a hack on our side but I'm not a fan of this idea. @mattrobenolt, thoughts?

**Comment by BYK** [comment]
I think this was fixed via getsentry/onpremise#1076 so closing.
